components/gitpod-protocol/src/protocol.ts

```diff
@@ -968,9 +977,47 @@ export namespace AdditionalContentContext {
     }
 }
 
-export interface CommitContext extends WorkspaceContext, Commit {
+export interface CommitContext extends WorkspaceContext, GitCheckoutInfo {
     /** @deprecated Moved to .repository.cloneUrl, left here for backwards-compatibility for old workspace contextes in the DB */
     cloneUrl?: string
+
+    /**
+     * The clone and checkout information for additional repositories in case of multi-repo projects.
+     */
+    additionalRepositoryCheckoutInfo?: GitCheckoutInfo[];
+}
+
+export namespace CommitContext {
+    /**
+     * Creates an identifier pointing to the commits of the CommitContext and all sub-repo commit infos.
+     * The identifier is max 255 chars long.
+     * @param commitContext
+     * @returns identifier for set of commits
+     */
+    export function getCommitSHAs(commitContext: CommitContext): string {
+        if (!commitContext.additionalRepositoryCheckoutInfo || commitContext.additionalRepositoryCheckoutInfo.length === 0) {
+            return commitContext.revision;
+        }
+        // it's a multi-repo project, we need to store all commit shas and that needs to fit into 255 chars. Let's only store at most first seven.
```

Comment:
I initially misread this as "let's remember only the first 7 additional repos / commits" 🙈

```suggestion
        // it's a multi-repo project, we need to store all commit shas and that needs to fit into 255 chars. Let's only store at most the first seven chars of each commit sha.
```